src/main/java/dk/cngroup/lentils/controller/ProgressController.java

```diff
@@ -102,12 +102,27 @@ public String teamProgress(final @RequestParam("teamId") Long teamId, final Mode
     @GetMapping(value = "/stage")
     public String stageProgress(final @RequestParam("cypherId") Long cypherId,
                                 final @RequestParam(value = "search", required = false) String searchString,
+                                final @RequestParam(value = "viewOtherTeams", required = false) Boolean viewOtherTeams,
                                 final Model model) {
         Cypher cypher = cypherService.getCypher(cypherId);
+        List<Team> searchedTeams = progressService.getSearchedTeams(searchString);
```

Comment:
Tohle neni dobre reseni. Nejprve se vyhledaji tymy (1. query) a pak pro kazdy tym se vola dalsi query (`existsStatusByCypherAndTeamAndCypherStatus`) a to dokonce dvakrat, pokud chceme videt i jine nez pending sifry. To znamena ze se dostavame nekam ke slozitosti `n` requestu pro `n` tymu. V praxi to pak bude presne `n + 1` a nebo `2n + 1` a to neni rozhodne idealni.
Idealni reseni je mit pouze jedno query, ktere nam vrati presne to, co potrebujeme. Nejlepsi zpusob jak toho dosahnout se mi jevi pouzit `Criteria` (Jednoduchy uvod, jak se to pouziva napriklad zde: https://www.baeldung.com/spring-data-criteria-queries)

Reply:
nakonec se mi to podari udelat pres ty specifikace, ale stejne tam budou 2 query, jedna pro aktivni tymy na sifre, druha pro neaktivni tymy na sifre